data-prepper-expression/src/main/antlr/DataPrepperStatement.g4

```diff
@@ -0,0 +1,262 @@
+
+fragment
+StringCharacters
+    : StringCharacter+
+    ;
+
+fragment
+StringCharacter
+    : ~["\\$]
+    | EscapeSequence
+    ;
+
+fragment
+EscapeSequence
+    : '\\' [btnfr"'\\$]
+    ;
+
+fragment
+JsonPointerEscapeSequence
+    : '\\' [btnfr"'\\/]
+    ;
+
+fragment
+JsonPointerStringCharacters
+    : JsonPointerStringCharacter+
+    ;
+
+fragment
+JsonPointerStringCharacter
+    : ~["\\]
+    | JsonPointerEscapeSequence
+    ;
+
+fragment
+JsonPointerCharacter
+    : [A-Za-z0-9_]
+    ;
+
+fragment
+JsonPointerCharacters
+    : JsonPointerCharacter+
+    ;
+
+JsonPointer
+    : DOUBLEQUOTE FORWARDSLASH JsonPointerStringCharacters? DOUBLEQUOTE
+    | FORWARDSLASH JsonPointerCharacters? (FORWARDSLASH JsonPointerCharacters)*
+    ;
+
+String
+    : DOUBLEQUOTE StringCharacters? DOUBLEQUOTE
+    ;
+
+fragment
+VariableNameLeadingCharacter
+    : [A-Za-z_]
+    ;
+
+fragment
+VariableNameCharacter
+    : VariableNameLeadingCharacter
+    | [0-9-]
+    ;
+
+fragment
+VariableNameCharacters
+    : VariableNameLeadingCharacter VariableNameCharacter*
+    ;
+
+VariableIdentifier
+    : '${' VariableNameCharacters '}'
+    ;
+
+statement
+    : expression EOF
+    | OTHER {System.err.println("unknown char: " + $OTHER.text);}
+    ;
+
+expression
+    : binaryOperatorExpression
+    ;
+
+binaryOperatorExpression
+    : conditionalExpression
+    ;
+
+conditionalExpression
+    : conditionalExpression conditionalOperator equalityOperatorExpression
+    | equalityOperatorExpression
+    ;
+
+equalityOperatorExpression
+    : equalityOperatorExpression binaryOperator regexOperatorExpression
+    | regexOperatorExpression
+    ;
+
+regexOperatorExpression
+    : regexOperatorExpression regexEqualityOperator regexPattern
+    | relationalOperatorExpression
+    ;
+
+relationalOperatorExpression
+    : relationalOperatorExpression relationalOperator setOperatorExpression
+    | setOperatorExpression
+    ;
+
+setOperatorExpression
+    : setOperatorExpression setOperator setInitializer
+    | unaryOperatorExpression
+    ;
+
+unaryOperatorExpression
+    : primary
+    | setInitializer
+    | regexPattern
+    | unaryNotOperatorExpression
+    ;
+
+unaryNotOperatorExpression
+    : unaryOperator primary
+    ;
+
+binaryOperator
+    : relationalOperator
+    | equalityOperator
+    ;
+
+regexEqualityOperator
+    : '=~'
+    | '!~'
+    ;
+
+setOperator
+    : 'in'
+    | 'not in'
+    ;
+
+
+conditionalOperator
+    : 'and'
+    | 'or'
+    ;
+
+unaryOperator
+    : 'not'
+    ;
+
+equalityOperator
+    : '=='
+    | '!='
+    ;
+
+relationalOperator
+    : '<'
+    | '<='
+    | '>'
+    | '>='
+    ;
+
+primary
+    : literal
+    | variableIdentifier
+    | setInitializer
+    | expressionInitializer
+    ;
+
+regexPattern
+    : JsonPointer
+    | String
+    | '(' (JsonPointer | String) ')'
+    ;
+
+expressionInitializer
+    : '(' expression ')'
+    ;
+
+setInitializer
+    : '{' binaryOperatorExpression (',' binaryOperatorExpression)* '}'
```

Comment:
Why does `setInitializer` take a `binaryOperatorExpressions` and not a `literal`?

Reply:
To reserve the possibility of a set of expressions. A possible use case might be `true in [/status_code == 500, /status_code == 404, /response_body == ""]`

Reply:
Another example we might want to support in the future would be `100 in {/status_code % 100, /status_code + 90}`. For now I'll build the grammar for lists to only support Literals or JsonPointers. We can update the grammar later if Data Prepper needs to support sets of statements.